cli/azd/pkg/httputil/util.go

```diff
@@ -25,3 +27,12 @@ func ReadRawResponse[T any](response *http.Response) (*T, error) {
 
 	return instance, nil
 }
+
+// Handles and errors executing the http request
+func HandleRequestError(response *http.Response, err error) error {
+	if response == nil {
```

Comment:
@wbreza @ellismg In the current case where `err != nil && response != nil`, we end up calling `NewResponseError` on `err`, which feels wrong.

I wonder we should use `err` only if `err != nil`. Not sure if this case can actually happen in reality.

Reply:
The case we are solving for is when `response == nil` && `err != nil`.  Here if the `response == nil` we'll just create an err from the `err`, otherwise we'll generate the HTTP response error from the specified response.